src/Template/Element/top_menu.ctp

```diff
@@ -32,6 +32,12 @@ $session = $this->request->getSession();
 $currentLanguage = $session->read('browse_sentences_in_lang');
 $showTranslationsInto = $session->read('show_translations_into_lang');
 
+$filteredLanguage = $session->read('vocabulary_requests_filtered_lang');
+
+if ($filteredLanguage == null) {
+    $filteredLanguage = 'und';
+}
```

Comment:
This if-block is not needed because the page contents will be the same wether the URL ends with `/und` or with nothing. It’s okay to leave `$filteredLanguage` as null.